lib/tbd/geo.rb

```diff
@@ -701,24 +701,40 @@ def convex?(s1 = nil, s2 = nil)
   # @param boundary ["Ground", "Foundation"] new outside boundary condition
   #
   # @return [Bool] true if model is free of KIVA-related objects
-  # @retrun [false] if invalid input (see logs)
+  # @return [false] if invalid input (see logs)
   def resetKIVA(model = nil, boundary = "Foundation")
     mth = "TBD::#{__callee__}"
     cl  = OpenStudio::Model::Model
     ck1 = model.is_a?(cl)
     ck2 = boundary.respond_to?(:to_s)
-    return mismatch("model"   , model   , cl    , mth, DBG, false) unless ck1
-    return mismatch("boundary", boundary, String, mth, DBG, false) unless ck2
+    kva = false
+    b   = ["Ground", "Foundation"]
+    return mismatch("model"   , model   , cl    , mth, DBG, kva) unless ck1
+    return mismatch("boundary", boundary, String, mth, DBG, kva) unless ck2
 
     boundary.capitalize!
-    ck3 = ["Ground", "Foundation"].include?(boundary)
-    return invalid("boundary", mth, 2, DBG, false) unless ck3
+    return invalid("boundary", mth, 2, DBG, kva) unless b.include?(boundary)
+
+    # Reset surface KIVA-related objects.
+    model.getSurfaces.each do |surface|
+      kva = true unless surface.adjacentFoundation.empty?
+      kva = true unless surface.surfacePropertyExposedFoundationPerimeter.empty?
+      surface.resetAdjacentFoundation
+      surface.resetSurfacePropertyExposedFoundationPerimeter
+      next unless surface.isGroundSurface
+      next unless surface.outsideBoundaryCondition.capitalize == boundary
+
+      lc = surface.construction.empty? ? nil : surface.construction.get
+      surface.setOutsideBoundaryCondition(boundary)
+      next if boundary == "Ground"
+      next if lc.nil?
+
+      surface.setConstruction(lc) if surface.construction.empty?
```

Comment:
Having to _re-_set a surface construction, AFTER setting "Foundation" as outside boundary condition, used to be necessary (maybe for v2.9.1). This no longer seems necessary, but keeping the solution as a redundant safeguard.